scrapy/pipelines/media.py

```diff
@@ -207,3 +219,7 @@ def item_completed(self, results, item, info):
                         extra={'spider': info.spider}
                     )
         return item
+
+    def file_path(self, request, response=None, info=None, item=None):
+        """Returns the path where downloaded media should be stored"""
+        pass
```

Comment:
Maybe `raise NotImplementedError`, just in case someone tries to use this class directly?

Reply:
We could do this, but `file_path` or `_file_path` is **not called anywhere** in the `MediaPipeline`, it is only called in its subclasses `ImagesPipeline` and `FilesPipeline` where `file_path` is implemented, so even if a user overrides or uses this class directly `file_path` will not be called. It looks like this class is not meant to be used directly by users according to the [docs](https://docs.scrapy.org/en/latest/topics/media-pipeline.html) and I'm guessing that's why its not implemented here. 

However if we raise an exception here, I think we should raise the exception in `media_to_download` and `get_media_requests` as well since they are also not implemented?

Reply:
Fair points.